pkg/backend/display/progress.go

```diff
@@ -342,6 +342,86 @@ func RenderProgressEvents(
 	close(done)
 }
 
+type CaptureProgressEvents struct {
+	Buffer  *bytes.Buffer
+	Stack   tokens.StackName
+	Proj    tokens.PackageName
+	display *ProgressDisplay
+}
+
+// NewCaptureProgressEvents is a wrapper around ProgressDisplay capturing
+// the output into an internal buffer and holding the display instance
+// for inspection after processing the events.
+//
+// E.g. It exposes whether a failure was detected in the display layer
+// and allows for inspection of the output.
+func NewCaptureProgressEvents(
+	stack tokens.StackName,
+	proj tokens.PackageName,
+	opts Options,
+) *CaptureProgressEvents {
+	buffer := bytes.NewBuffer([]byte{})
+	width, height := 200, 80
```

Comment:
Not sure I'm groking all this code yet, but having these hardcoded numbers here seems a little strange.